TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/XDrive.java

```diff
@@ -0,0 +1,187 @@
+import org.firstinspires.ftc.robotcore.external.navigation.AxesOrder;
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.Quaternion;
+import org.firstinspires.ftc.teamcode.core.lib.gamepad.GamepadManager;
+import org.firstinspires.ftc.teamcode.core.lib.gamepad.SmartGamepad;
+import org.firstinspires.ftc.teamcode.core.lib.interfaces.Subsystem;
+import org.firstinspires.ftc.teamcode.core.util.MathUtils;
+
+import java.util.HashMap;
+
+public class XDrive implements Subsystem {
+
+    private  Telemetry telemetry;
+    private static XDrive instance;
+    private double reset_angle = 0;
+    private DcMotor front_left = null;
+    private DcMotor back_left = null;
+    private DcMotor back_right = null;
+    private DcMotor front_right = null;
+    private IMU imu;
+    private float imuRoll = 0;
+    private float imuPitch = 0;
+    private float imuYaw = 180;
+
+    private XDrive() {
+    }
+
+    @Override
+    public void initialize(HardwareMap hardwareMap, Telemetry telemetry) {
+        front_left = hardwareMap.dcMotor.get(MOTOR_FRONT_LEFT);
+        back_left = hardwareMap.dcMotor.get(MOTOR_BACK_LEFT);
+        back_right = hardwareMap.dcMotor.get(MOTOR_BACK_RIGHT);
+        front_right = hardwareMap.dcMotor.get(MOTOR_FRONT_RIGHT);
+
+        front_left.setDirection(DcMotor.Direction.REVERSE);
+        back_left.setDirection(DcMotor.Direction.REVERSE);
+        front_right.setDirection(DcMotor.Direction.FORWARD);
+        back_right.setDirection(DcMotor.Direction.FORWARD);
+
+        front_left.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        back_left.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        front_right.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        back_right.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+
+        this.telemetry = telemetry;
+
+        imu = hardwareMap.get(IMU.class, "imu");
+        IMU.Parameters parameters = new IMU.Parameters(
+                new RevHubOrientationOnRobot(
+                        RevHubOrientationOnRobot.LogoFacingDirection.DOWN,
+                        RevHubOrientationOnRobot.UsbFacingDirection.FORWARD
+                )
+        );
+        imu.initialize(parameters);
+    }
+
+    @Override
+    public void start() {
+    }
+
+    @Override
+    public void stop() {
+    }
+
+    @Override
+    public void execute(GamepadManager gamepadManager) {
+        drive(gamepadManager.getDriver());
+        resetAngle(gamepadManager.getDriver());
+    }
+
+    public void drive(SmartGamepad driver) {
+        double rotate = driver.getRightStickX()/4;
+        double stick_x = driver.getLeftStickX() * Math.sqrt(Math.pow(1-Math.abs(rotate), 2)/2);
+        double stick_y = driver.getLeftStickY() * Math.sqrt(Math.pow(1-Math.abs(rotate), 2)/2);
+        double theta = 0;
+        double Px = 0;
+        double Py = 0;
```

Comment:
variable name should always start with lower case + use var to avoid verbose code

Reply:
Same problem, var does't work